airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -13,6 +16,49 @@ const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
   return streamNode.config?.cursorField || [];
 };
 
+const clearBreakingFieldChanges = (nodeStream: SyncSchemaStream, breakingChangesByStream: StreamTransform[]) => {
+  if (!breakingChangesByStream?.length) {
+    return nodeStream;
+  }
+
+  let clearPrimaryKey = false;
+  let clearCursorField = false;
+
+  for (const streamTransformation of breakingChangesByStream) {
+    // get all of the removed field paths for this transformation
+    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
+
+    if (!removedFieldPaths?.length) {
+      return nodeStream;
+    }
+
+    // if there is a primary key in the config, and any of its field paths were removed, we'll be clearing it
+    if (
+      !!nodeStream.config?.primaryKey?.length &&
+      nodeStream.config?.primaryKey?.some((key) => removedFieldPaths.some((removedPath) => isEqual(key, removedPath)))
+    ) {
+      clearPrimaryKey = true;
+    }
```

Comment:
Yes.  Because we _only_ want to override for the `true` case, not for `false` (which could mistakenly overwrite a previous item setting the variable as true), it would end up reading something like `clearPrimaryKey=!!(!!nodeStream.config?.primaryKey?.length && ....` which did not seem very readable.